### PR TITLE
Switch to asciiart for boxplot illustration.

### PR DESCRIPTION
**Pull request by anntzer**
This makes the information also available at the terminal.

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Comment by timhoffm** [comment]
From https://github.com/matplotlib/matplotlib/pull/19860#issuecomment-813107087

> I'm -0.1 on this because the ascii-art is only a rough visual approximation of the box and harder to read than an actual image. It's a trade-off between having the info in the ascii docstring and readability in the html output. But I won't block a PR.
> 
> Note that if you do the ascii-art, you'll likely have to use some left-side character for aligning the ascii lines. ReST messes up if any line in the code block is less indented than the first line. See e.g. https://matplotlib.org/devdocs/api/_as_gen/matplotlib.axes.Axes.barbs.html

**Comment by anntzer** [comment]
fwiw I think this renders just fine on the website
bxp
but let's see what others think...
(perhaps I can use "o" instead of "." for outliers, though)

**Comment by jklymak** [comment]
Personally I would remove *most* of this verbiage, and just point to wikipedia and an example (assuming we have one).  I agree this didn't belong in the FAQ, but I don't think it really belongs in the docstring either...

**Comment by timhoffm** [comment]
I'm fine with that as well.

**Comment by anntzer** [comment]
I got rid of most of the verbiage, but kept the asciiart diagram and moved it to the top.  I don't think it takes too much space there?
I also harmonized "box plot" to "boxplot" in the docstring.

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -3550,26 +3550,38 @@ def boxplot(self, x, notch=None, sym=None, vert=None, whis=None,
...
+        quartile (Q3) of the data, with a line at the median.  The
+        whiskers extend from the box by 1.5x the inter-quartile range
+        (IQR).  Flier points are those past the end of the whiskers.
+        See https://en.wikipedia.org/wiki/Box_plot for reference.
+
+        .. code-block:: none
+
+                  Q1-1.5IQR   Q1   median  Q3   Q3+1.5IQR
+                               |-----:-----|
+               .      |--------|     :     |--------|    .  .
```
Comment (story645) [inline lib/matplotlib/axes/_axes.py:3565]: agree w/ you on 'O'

**Review by story645 — APPROVED** [review]
I think the ASCII art is kinda brilliant - is illustrative and to the point and I think worth it here since odds are most folk won't look at wikipedia.

File: lib/matplotlib/axes/_axes.py (hunk 1 above)
Comment (anntzer) [reply lib/matplotlib/axes/_axes.py]: "o" it is

**Pull request merged**